lib/spack/spack/binary_distribution.py

```diff
@@ -241,7 +241,7 @@ def build_tarball(spec, outdir, force=False, rel=False, yes_to_all=False,
         else:
             raise NoOverwriteException(str(specfile_path))
     # make a copy of the install directory to work with
-    workdir = join_path(outdir, os.path.basename(spec.prefix))
+    workdir = join_path(outdir, spec.dag_hash())
```

Comment:
As a sanity check, this change is not strictly required right? In other words, it is not the problem you refer to in https://github.com/spack/spack/pull/6510#issue-277587014: 

> Unpacking a tarball is a spack install with a different directory layout caused errors because the basename of the prefix directory was included in the tarball. Removing the prefix directory fixes the problem.

Reply:
Correct. This was just a convenience for me.